rewrite-core/src/main/java/org/openrewrite/marker/Marker.java

```diff
@@ -18,8 +18,20 @@
 import com.fasterxml.jackson.annotation.JsonIdentityInfo;
 import com.fasterxml.jackson.annotation.JsonTypeInfo;
 import com.fasterxml.jackson.annotation.ObjectIdGenerators;
+import org.openrewrite.Tree;
+import org.openrewrite.TreePrinter;
+import org.openrewrite.TreeVisitor;
 
 @JsonIdentityInfo(generator = ObjectIdGenerators.IntSequenceGenerator.class, property = "@ref")
 @JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, property = "@c")
-public interface Marker {
+public interface Marker extends Tree {
+    @Override
+    default <P> boolean isAcceptable(TreeVisitor<?, P> v, P p) {
+        return false;
```

Comment:
Unsure about this...

Reply:
`Marker` can extend `Tree` all we want. But the `T` in  `TreeVisitor<T extends Tree, P>` always constrains visitors to exactly one sub-type of `Tree`.  For example, `JavaVisitor` visits `J`, not `Marker`.

We could get rid of the type parameter in `TreeVisitor` and do more casting everywhere. That feels like a step backwards in API usability and clarity for every other, non-Marker related scenario.

If we're not changing the type constraints on `TreeVisitor` then it follows that even though `Marker` implements `Tree`, it doesn't accept `JavaVisitor`s, because `JavaVisitor.visit(Marker)` is going to have to try and cast that `Marker` to a `J` and fail with a `ClassCastException`.

Reply:
We could create language-specific Markers... so there would be a `J.Marker` would accept `JavaVisitor` and such. Creating not just `RecipeSearchResult` but also `JavaSearchResult` might be a step in that direction. That might make language-agnostic markers like `GitProvenance` awkward.